packages/contracts/contracts/ERC2981ContractWideRoyalties.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import '@openzeppelin/contracts/utils/introspection/ERC165.sol';
+
+import './ERC2981Base.sol';
+
+/// @dev This is a contract used to add ERC2981 support to ERC721 and 1155
+/// @dev This implementation has the same royalties for each and every tokens
+abstract contract ERC2981ContractWideRoyalties is ERC2981Base {
+    RoyaltyInfo private _royalties;
+
+    /// @dev Sets token royalties
+    /// @param recipient recipient of the royalties
+    /// @param value percentage (using 2 decimals - 10000 = 100, 0 = 0)
+    function _setRoyalties(address recipient, uint256 value) internal {
+        require(value <= 10000, 'ERC2981Royalties: Too high');
+        _royalties = RoyaltyInfo(recipient, uint24(value));
+    }
+
+    /// @inheritdoc	IERC2981Royalties
+    function royaltyInfo(uint256, uint256 value)
```

Comment:
is this not a compilation error?

Reply:
Actually, no. the actual argument is a tokenID. We can implement royalties per token but I believe our usecase is the same royalty for all Passports within a membership type. Since we donot use tokenId inside the function, this way of defining the function works. I have dev-tested the contract as well.

Reply:
Actually, no. the actual argument is a tokenID. We can implement royalties per token but I believe our usecase is the same royalty for all Passports within a membership type. Since we donot use tokenId inside the function, this way of defining the function works. I have dev-tested the contract as well.

Reply:
wow! today I learned